src/browsers/tob/shared/components/DotplotHeatmap.js

```diff
@@ -0,0 +1,241 @@
+        <g transform={`translate(${margin.left}, ${margin.top})`}>
+          <g id="x-axis">
+            <line x2={`${innerWidth}`} stroke="black" />
+            <line y1={`${innerHeight}`} y2={`${innerHeight}`} x2={`${innerWidth}`} stroke="black" />
+            {xScaleLocal.domain().map((tick) => (
+              <g
+                key={tick}
+                transform={`translate(${
+                  xScaleLocal(tick) + xScaleLocal.bandwidth() / 2
+                }, ${innerHeight})`}
+              >
+                <text
+                  transform="translate(0, 10)rotate(-90)"
+                  style={{ textAnchor: 'end', alignmentBaseline: 'middle' }}
+                >
+                  {tick}
+                </text>
+                <line y2={6} stroke="black" />
+                <line y2={`-${innerHeight}`} stroke="lightgrey" />
+              </g>
+            ))}
+          </g>
+          <g id="y-axis">
+            <line y2={`${innerHeight}`} stroke="black" />
+            <line x1={`${innerWidth}`} y2={`${innerHeight}`} x2={`${innerWidth}`} stroke="black" />
+            {yScaleLocal.domain().map((tick) => (
+              <g
+                key={tick}
+                transform={`translate(0, ${yScaleLocal(tick) + yScaleLocal.bandwidth() / 2})`}
+              >
+                <text key={tick} style={{ textAnchor: 'end', alignmentBaseline: 'middle' }} x={-6}>
+                  {tick}
+                </text>
+                <line x2={-3} stroke="black" />
+                <line x2={`${innerWidth}`} stroke="lightgrey" />
+              </g>
+            ))}
+          </g>
+          <g id="tiles">
+            {data.length ? (
+              data.map((item) => (
+                <circle
+                  key={`${accessors.x(item)},${accessors.y(item)}`}
+                  r={sizeScaleLocal(accessors.size(item))}
+                  cx={xScaleLocal(accessors.x(item)) + xScaleLocal.bandwidth() / 2}
+                  cy={yScaleLocal(accessors.y(item)) + yScaleLocal.bandwidth() / 2}
+                  stroke="black"
+                  fill={`${colorScaleLocal(accessors.color(item))}`}
+                  onClick={onClick}
+                />
+              ))
+            ) : (
+              <g transform={`translate(${innerWidth / 2}, ${innerHeight / 2})`}>
+                <text
+                  style={{ textAnchor: 'middle', alignmentBaseline: 'middle', fontSize: 'larger' }}
+                >
+                  No data to display.
+                </text>
+              </g>
+            )}
+          </g>
+          {data.length && (
+            <g
+              id="sizeLegend"
+              transform={`translate(${
+                width - margin.right - margin.left + sizeScaleLocal.range()[1] + 5
+              }, 20)`}
+            >
+              <text fontWeight="bold"> P value </text> <br />
+              {sizeScaleLocal
+                .ticks(6)
+                .filter((size) => size < 1)
```

Comment:
Why are the ticks filtered for `size < 1`? Applying a log transform means most ticks will have size > 1 so it's better not to make assumptions on the nature of the data by applying a filter.

Reply:
Relic from when p-value was between 0 and 1. A pvalue of 1 had a circle radius of 0 so wouldn't show up in the scale so I omitted it.

Reply:
Might need to revisit this later